release-notes/7.0/preview/api-diff/preview3/Microsoft.AspNetCore.App/7.0-preview3_Microsoft.Extensions.Hosting.md

```diff
@@ -0,0 +1,46 @@
+# Microsoft.Extensions.Hosting
+
+``` diff
+ namespace Microsoft.Extensions.Hosting {
+     public abstract class BackgroundService : IDisposable, IHostedService {
+-        public virtual Task ExecuteTask { get; }
++        public virtual Task? ExecuteTask { get; }
+     }
+     public static class GenericHostBuilderExtensions {
++        public static IHostBuilder ConfigureWebHostDefaults(this IHostBuilder builder, Action<IWebHostBuilder> configure, Action<WebHostBuilderOptions> configureOptions);
+     }
+     public static class Host {
++        public static HostApplicationBuilder CreateApplicationBuilder();
++        public static HostApplicationBuilder CreateApplicationBuilder(string[] args);
+     }
++    public sealed class HostApplicationBuilder {
++        public HostApplicationBuilder();
++        public HostApplicationBuilder(HostApplicationBuilderSettings settings);
++        public HostApplicationBuilder(string[] args);
++        public ConfigurationManager Configuration { get; }
++        public IHostEnvironment Environment { get; }
++        public ILoggingBuilder Logging { get; }
++        public IServiceCollection Services { get; }
++        public IHost Build();
++        public void ConfigureContainer<TContainerBuilder>(IServiceProviderFactory<TContainerBuilder> factory, Action<TContainerBuilder> configure = null) where TContainerBuilder : notnull;
++    }
++    public sealed class HostApplicationBuilderSettings {
++        public HostApplicationBuilderSettings();
++        public string ApplicationName { get; set; }
++        public string[] Args { get; set; }
++        public ConfigurationManager Configuration { get; set; }
++        public string ContentRootPath { get; set; }
++        public bool DisableDefaults { get; set; }
++        public string EnvironmentName { get; set; }
++    }
+     public interface IHostEnvironment {
+-        string ApplicationName { get; set; }
++        string? ApplicationName { get; set; }
```

Comment:
Just a heads up. I think this should be non-nullable, but we do need to figure out what this gets set to when `Assembly.GetEntryAssembly()` returns null. I think this is fine for previews, but we should try to fix this for the final release of .NET 7. I filed an issue: #68512.